### PR TITLE
Add support for float and double

### PR DESCRIPTION
**Pull request by lukipuki**

### CODE REVIEW

**Comment by lukipuki** [comment]
Review recommended with hidden whitespace changes.

File: test/test.js [hunk 1]
```diff
@@ -13,43 +13,48 @@ const addAsync = {};
 const addTwiceAsync = {};
 let getpid;
 
-const sizes = [8, 16, 32, 64];
-
 const libAdder = path.join(__dirname, 'adder', 'libadder.so');
 
+var types = ['float', 'double'];
```
Comment (ppershing) [inline test/test.js:18]: wouldn't it be better if this was an array of structs {typ, size}?

File: test/test.js [hunk 2]
```diff
@@ -64,11 +69,13 @@ test`getBufferPointer`(() => {
   assert(testBufPtr > 0n);
 });
 
-function num (size, n) {
-  return size === 64 ? BigInt(n) : n;
+function num (size, typ, n) {
```
Comment (ppershing) [inline test/test.js:72]: it would make more sense to have typ as first argument

File: test/test.js (hunk 1 above)
Comment (lukipuki) [reply test/test.js:18]: Sure, I was just a little lazy :)

File: test/test.js [hunk 3]
```diff
@@ -13,43 +13,48 @@ const addAsync = {};
 const addTwiceAsync = {};
 let getpid;
 
-const sizes = [8, 16, 32, 64];
-
 const libAdder = path.join(__dirname, 'adder', 'libadder.so');
 
+var types = ['float', 'double'];
+var sizes = [32, 64];
```
Comment (tamasbogi) [inline test/test.js:19]: try to avoid using `var`, use `const` or `let` instead

File: test/test.js (hunk 1 above)
Comment (lukipuki) [reply test/test.js]: Done

File: test/test.js (hunk 2 above)
Comment (lukipuki) [reply test/test.js]: Done

File: test/test.js (hunk 3 above)
Comment (lukipuki) [reply test/test.js]: Done

**Pull request merged**